modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerRecoveryTest.java

```diff
@@ -0,0 +1,131 @@
+import java.util.stream.Stream;
+import org.apache.ignite.internal.hlc.HybridClock;
+import org.apache.ignite.internal.hlc.HybridClockImpl;
+import org.apache.ignite.internal.hlc.HybridTimestamp;
+import org.apache.ignite.internal.manager.IgniteComponent;
+import org.apache.ignite.internal.metastorage.MetaStorageManager;
+import org.apache.ignite.internal.metastorage.impl.StandaloneMetaStorageManager;
+import org.apache.ignite.internal.metastorage.server.KeyValueStorage;
+import org.apache.ignite.internal.metastorage.server.TestRocksDbKeyValueStorage;
+import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
+import org.apache.ignite.internal.testframework.WorkDirectory;
+import org.apache.ignite.internal.testframework.WorkDirectoryExtension;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.apache.ignite.internal.vault.VaultManager;
+import org.apache.ignite.internal.vault.inmemory.InMemoryVaultService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/** For {@link CatalogManager} testing on recovery. */
+@ExtendWith(WorkDirectoryExtension.class)
+public class CatalogManagerRecoveryTest extends BaseIgniteAbstractTest {
+    private static final String NODE_NAME = "test-node-name";
+
+    @WorkDirectory
+    private Path workDir;
+
+    private final HybridClock clock = new HybridClockImpl();
+
+    private final VaultManager vaultManager = new VaultManager(new InMemoryVaultService());
+
+    private MetaStorageManager metaStorageManager;
+
+    private CatalogManager catalogManager;
+
+    @AfterEach
+    void tearDown() throws Exception {
+        IgniteUtils.stopAll(catalogManager, metaStorageManager, vaultManager);
+    }
+
+    @Test
+    void testRecoveryCatalogVersionTimestamps() throws Exception {
+        createAndStartComponents();
+
+        // Let's create a couple of versions of the catalog.
+        assertThat(catalogManager.execute(simpleTable(TABLE_NAME)), willCompleteSuccessfully());
+
+        int catalogVersion0 = catalogManager.latestCatalogVersion();
+
+        assertThat(catalogManager.execute(simpleIndex(TABLE_NAME, INDEX_NAME)), willCompleteSuccessfully());
+
+        int catalogVersion1 = catalogManager.latestCatalogVersion();
+
+        assertThat(catalogVersion1, greaterThan(catalogVersion0));
+
+        long time0 = catalogManager.catalog(catalogVersion0).time();
+        long time1 = catalogManager.catalog(catalogVersion1).time();
+
+        assertThat(time1, greaterThan(time0));
+
+        // We will restart and recover the components and also set the clock to the future.
+        stopComponents();
+
+        HybridTimestamp updateNow = clock.now().addPhysicalTime(TimeUnit.DAYS.toMillis(1));
+        clock.update(updateNow);
+
+        HybridTimestamp newNow = clock.now();
+        assertTrue(newNow.compareTo(updateNow) > 0, "updateNow=" + updateNow + ", newNow=" + newNow);
```

Comment:
Can `assertThat(newNow, greaterThan(updateNow))` be used here?

Reply:
why not